src/pages/assets/AssetsTable.js

```diff
@@ -45,28 +49,60 @@ const AssetsTable = ({
   refetchAssets,
   setMinVolumeFilter,
   minVolume = 10000,
-  listName
+  listName,
+  settings,
+  setHiddenColumns
 }) => {
-  const { isLoading, error } = Assets
+  const { isLoading, error, timestamp, typeInfo } = Assets
+  const key = typeInfo.listId || listName
+  const { sorting, pageSize, hiddenColumns } = settings[key] || {}
   if (error && error.message !== 'Network error: Failed to fetch') {
     return <ServerErrorMessage />
   }
 
-  const [columnsSettings, changeColumnsSettings] = useState(
-    columnSettingsDefault
+  const changeShowing = (columns, hiddenColumns) => {
+    const modifiedColumns = JSON.parse(JSON.stringify(columns))
+    hiddenColumns.forEach(name => (modifiedColumns[name].show = false))
+
+    return modifiedColumns
+  }
+
+  const savedHidden = hiddenColumns || COMMON_SETTINGS.hiddenColumns
+  const sortingColumn = sorting || COMMON_SETTINGS.sorting
+  const columnsAmount = pageSize || COMMON_SETTINGS.pageSize
+
+  const [hidden, changeHidden] = useState(savedHidden)
+  const [columns, changeColumns] = useState(
+    changeShowing(COLUMNS_SETTINGS, savedHidden)
   )
 
-  const toggleColumn = ({ id, show, selectable }) => {
-    const toggledColumns = Object.assign({}, columnsSettings)
-    toggledColumns[id] = {
-      ...toggledColumns[id],
+  if (savedHidden !== hidden) {
+    changeHidden(savedHidden)
+    changeColumns(changeShowing(COLUMNS_SETTINGS, savedHidden))
+  }
+
+  const toggleColumn = ({ name, show, selectable }) => {
+    const toggledColumns = Object.assign({}, columns)
+    // NOTE(haritonasty): such access to the fields is necessary for Safari bug (shuffle properties)
+    toggledColumns[name] = {
+      ...toggledColumns[name],
       show: selectable ? !show : show
```

Comment:
@haritonasty why not show: !selectable

Reply:
we don't wan't toggle disabled items: 
![image](https://user-images.githubusercontent.com/24521041/62131808-91ce1300-b2e4-11e9-91f7-bc7b14d2320b.png)